kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -60,12 +65,18 @@ public class TableConfig<T> {
             "needs to be a positive integer."
     );
 
+    private static final HashMap<String, TableConfig> entries = new HashMap<>();
```

Comment:
What is entries? The set of all legal properties? Please use a more descriptive name and add a docstring as well.